### DISCUSSION TITLE
Read font from URL

### DISCUSSION THREAD

**Discussion by gTutini** [schoenbergerb/noscrape#44] [Ideas]
Hi!
First, I want to thank you for the splendid work on this lib! It's a awesome idea and already helping me a lot with my job.
And, I want to ask if there's a possibility of reading fonts from a direct file URL (something like `https://mydomain.com/font.woff2`) in order to use this with client side only, to usage with React Apps with Vite, for example.
If it's possible, I want to contribute with this change!

Again, thanks a lot o/

**Comment by schoenbergerb** [comment]
Hi Giancarlo,

I had a quick look at that, and yes, it would be possible.

The currently used `loadSync` function provided by `opentype.js` works only with local file paths. However, one could use the provided `parse` function, which accepts an ArrayBuffer. Thus, one could use either the fs or fetch API to load the file.

The question is, should `noscrape` handle the differentiation between a "remote URL" and a "local file path"?

I tend not to, because it seems a little error-prone to me. I asked ChatGPT, and it gave me just a few weird regular expressions, and I think it's more reasonable to leave the part of loading the file in the developers' hands. They usually know where the fonts come from and what APIs are available to load them 😅.

But this means one has to change the constructor, something like that:
``` typescript
  constructor(
    fontBuffer: ArrayBuffer | SharedArrayBuffer,
    options?: ObfuscationOptions,
  ) {
    this.options = {
      ...DEFAULT_OPTIONS,
      ...options,
    };

    this.font = parse(fontBuffer);
  }
```

What do you think?

**Reply by schoenbergerb** [reply]
btw, thank you for your warm words, it's nice to hear that someone is using this and the work was worth it ☺️

**Reply by gTutini** [reply]
Hi! Thanks for the response!
That looks awesome! It solves the problem and it's not a big change, from what it looks like. A very good solution 💪
For compatibility reasons, maybe a option can set if the constructor gets a filepath directly or the new array buffer, what do you think?

Again, thanks for the good solution and the fast response o/

**Reply by schoenbergerb** [reply]
yo, you are right, that makes sense because we wouldn't have a breaking change then

**Comment by schoenbergerb** [comment]
Resolved, from Version 2.1.0 onwards, one can instantiate Noscrape either by a local file path (string) or an ArrayBuffer. 
I recommend v2.1.1; I made a mistake by including Playwright as a new test engine 🙈.

**Reply by gTutiniOrla** [reply]
Many thanks again!

**Reply by schoenbergerb** [reply]
you are welcome :)
